Refactoring

margineanmihai opened the pull request:
Refactored Controllers, DB connection, Authentication
Added Routes, Models
Fixed Header bugs

Commit:
Refactoring
Controllers, Models, Routes, DB connection, Authentication
Fixed Header bugs

skylock commented on app.js:
```diff
@@ -1,21 +1,38 @@
+
+/* MONGOOSE SETUP */
+const mongoose = require('mongoose');
+mongoose.connect('mongodb+srv://mihai:mihai@cluster0.j9czx.mongodb.net/musicapp?retryWrites=true&w=majority',{ 
```
Picking up secrets from the environment is preferred. Committing to the repo should be avoided

margineanmihai replied:
Thank you for mentioning that. Fixed it

skylock reviewed (approved):
Nice work, although smaller commits would make the review process easier.

Commit:
Added eslint, fixed eslint problems

Commit:
Implement env var

margineanmihai commented:
> Nice work, although smaller commits would make the review process easier.

Thank you and you're right. I should have done smaller commits over time.

skylock commented on app.js:
```diff
@@ -22,7 +23,7 @@ app.use(routes);
 /* MONGOOSE SETUP */
 const mongoose = require("mongoose");
-mongoose.connect("mongodb+srv://mihai:mihai@cluster0.j9czx.mongodb.net/musicapp?retryWrites=true&w=majority",{ 
+mongoose.connect(`${process.env.DB_HOST}://${process.env.DB_USER}:${process.env.DB_PASS}@cluster0.j9czx.mongodb.net/${process.env.DB_NAME}?retryWrites=true&w=majority`,{ 
```
Nice work !

Radu94 commented on app.js:
```diff
@@ -1,21 +1,39 @@
-loginController(app);
-const passport = require('passport');
+require("dotenv").config();
+const express=require("express");
```
spacing is a bit off :) but not the worse thing in the world :)

Radu94 commented on controllers/trackController.js:
```diff
@@ -1,131 +1,87 @@
+    });       
+};
+
+exports.cartDeleteItem = (req, res) => {
```
try to be consistent with your exports :)
as a personal preference i usually export everything i need to on the last line, it makes everything easier to read :)

Commit:
Formating with prettier

Commit:
Use module.exports instead of exports
Have only one export instead of multiple ones
Update eslint and prettier configs
Update gitignore
